ios/src/TGMapViewController.mm

```diff
@@ -178,7 +178,7 @@ - (void)pickFeatureAt:(CGPoint)screenPosition
     screenPosition.y *= self.contentScaleFactor;
 
     self.map->pickFeatureAt(screenPosition.x, screenPosition.y, [screenPosition, self](const Tangram::FeaturePickResult* featureResult) {
-        if (!self.mapViewDelegate && ![self.mapViewDelegate respondsToSelector:@selector(mapView:didSelectFeature:atScreenPosition:)]) {
+        if (!self.mapViewDelegate || (self.mapViewDelegate && ![self.mapViewDelegate respondsToSelector:@selector(mapView:didSelectFeature:atScreenPosition:)])) {
```

Comment:
You can probably just switch these checks to be straight OR checks, right? Because if either one of these is nil (or both), you want to return. Since It's evaluated left to right, when `!self.mapViewDelegate` is nil, it'll set the if to true and return, without evaluating the secondary condition.

Basically:
`if (!self.mapViewDelegate || ![self.mapViewDelegate respondsToSelector:@selector(mapView:didSelectFeature:atScreenPosition:)])`